issue #2922 (#3002)
ElektraInitiative/libelektra: doc/GETSTARTED.md

# Get Started

- Skill requirements

>  - Working with  Linux

Reviewer: MacOS and BSDs are also fully supported. Windows might be supported but it is not tested much.

   one of the common linux distributions like [Ubuntu](https://ubuntu.com/download/desktop#download), 
   [Linux Mint](https://www.linuxmint.com/download.php), [openSuse](https://software.opensuse.org/distributions/leap) or 
   [Fedora](https://getfedora.org/en/). 
 - Using command-line interface and commands
 
   The easiest way to compile, install and use Elektra is by using the terminal. We will introduce the basic commands which you will need to run Elektra for the very first time. 
 - Basic knowledge about git
 
   Dont panic! [GIT](https://git-scm.com/) is a distributed version constrol system to track changes of the source code in a project. We will use a single command of GIT to get 
   the source code of Elektra. 
 - Basic knowledge about make/cmake
 
   Never heard about [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/), it’s okey! We will use this build-management tools to compile, 
   link dependencies etc. to build a executable program from the source code.  
 - Furthermore, a programming/scripting language like C, C++, ruby, JavaScript etc.
 

>    You will need a programming language to fulfill your needs

Reviewer: ?

 
- Software requirements

>    We need to install some basic tools to run Elektra: cmake3 , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively ninja and clang are 

Reviewer: Spaces? cmake3 looks like distribution-specific.

   also supported but not described here). Depending on your linux distribution use following commands to install these tools:
   ```sh
   sudo apt-get install cmake git build-essential
   ```
   Or on RPM (Red Hat Package Manager) based systems (like Fedora, openSuse, CentOS etc.):
   ```sh
   sudo yum install -y cmake3 git gcc-c++ 
   ```
   Or on macOS Sierra, most of the build tools can be obtained by installing Xcode (from the App Store). Other required tools may be installed using [brew ](https://brew.sh/). First install brew as 
   described on their website. Then issue the following command to get cmake to complete the basic requirements:
   ```sh
   brew install cmake git 
   ```
- Installation

  If you meet all of the software requirements you can get the source code of Elektra by using this command:
  ```sh
   git clone https://github.com/ElektraInitiative/libelektra.git 
   ```
   Run the following commands to compile Elektra with non-experimental parts where your system happens to fulfil the dependences:
   ```sh

>    cd libelektra

Reviewer: Here is a lot of overlap to doc/COMPILE.md

   mkdir build
   cd build
   cmake ..  # watch output to see if everything needed is included
   ccmake .. # optional: overview of the available build settings (needs cmake-curses-gui)
   make -j 5
   make run_nokdbtests  # optional: run tests
   ```
   
    After you completed building Elektra on your own, you can execute these commands to install Elektra:
    ```sh
    sudo make install
    sudo ldconfig
   ```
- Hello World!

  Start with your very first Elektra application and follow these steps:

>   [Hello World!](doc/tutorials/hello-elektra.md)

Reviewer: Link broken, absolute link starts with /.

I am not sure if your tutorial covered everything that COMPILE.md covered?

# Get Started

- Skill requirements

  - Operating system

>     We will need a unix-based operating system to run Elektra (Linux, BSD, macOS) but it's also possible to use Windows which is supported but not tested much. If you are new to the world of unix we recommend you to use one of the common linux distributions like [Ubuntu](https://ubuntu.com/download/desktop#download), [Linux Mint](https://www.linuxmint.com/download.php), [openSuse](https://software.opensuse.org/distributions/leap) or [Fedora](https://getfedora.org/en/).

Reviewer 2: ```suggestion
    We recommend a unix-based operating system to run Elektra (Linux, BSD, macOS) but it's also possible to use Windows which is supported but not yet fully tested.
```

>     We will need a unix-based operating system to run Elektra (Linux, BSD, macOS) but it's also possible to use Windows which is supported but not tested much. If you are new to the world of unix we recommend you to use one of the common linux distributions like [Ubuntu](https://ubuntu.com/download/desktop#download), [Linux Mint](https://www.linuxmint.com/download.php), [openSuse](https://software.opensuse.org/distributions/leap) or [Fedora](https://getfedora.org/en/).

Reviewer 2: If someone really wants to contribute to Elektra and does not have a unix based system, he will have severe problems learning unix based systems before actually contributing to Elektra. If someone already has experience with unix (which I assume the vast majority of Dev's contributing to C related projects have), these sentences are irrelevant. Therefore I would simply omit those sentences

  - Using command-line interface and commands

    The easiest way to compile, install and use Elektra is by using the terminal. We will introduce the basic commands which you will need to run Elektra for the very first time.

  - Basic knowledge about git

>     Dont panic! [GIT](https://git-scm.com/) is a distributed version constrol system to track changes of the source code in a project. We will use a single command of GIT to get

Reviewer 2: I like this sort of writing but I think that those sentences here are not needed. If somebody wants to contribute to a repository on github he will for sure know about git. Just link git website in the "Basic knowledge about git"

Reviewer: I also like the style. Maybe some students really do not know git, so I think it is okay to have it (maybe once in the whole tutorial)

    the source code of Elektra.

  - Basic knowledge about make/cmake

>     Never heard about [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/), it’s okey! We will use this build-management tools to compile, link dependencies etc. to build a executable program from the source code.

Reviewer 2: If someone really did not hear about make or cmake he will also have troubles understanding "link dependencies" (Depite I know both tools for example I do not exactly know what happens on dependency linkage).

Reviewer: If you will have troubles with cmake depends on what you want to do. If you only write a simple plugin, there should be no problem.

>     Never heard about [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/), it’s okey! We will use this build-management tools to compile, link dependencies etc. to build a executable program from the source code.

Reviewer 2: ```suggestion
    [make](https://www.gnu.org/software/make/) or [cmake](https://cmake.org/) are used to generate an executable program from the code. If you are not used to these tools its no problem, we will introduce them to you in latter sections.
```

Reviewer: latter -> later?

>   - (optional) Furthermore, a programming/scripting language like C, C++, ruby, JavaScript etc. ...

Reviewer 2: I dont think that this is needed as it is obvious

Reviewer: I think it is nice to also invite non-programmers to Elektra, e.g. for docu, website, translation,....

    You can contribute to Elektra to improve the source code, website, documentation, translation etc.

- Software requirements

>   We need to install some basic tools to run Elektra: cmake , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively [ninja](https://ninja-build.org/) and [clang](https://clang.llvm.org/index.html) are also supported but not described here). Depending on your linux distribution use following commands to install these tools:

Reviewer: ```suggestion
  We need to install some basic tools to run Elektra: cmake, git, and essential build tools (make, gcc, and some standard Unix tools; alternatively [ninja](https://ninja-build.org/) and [clang](https://clang.llvm.org/index.html) are also supported but not described here). Depending on your linux distribution use following commands to install these tools:
```

>   We need to install some basic tools to run Elektra: cmake , git , and essential build tools (make, gcc, and some standard Unix tools; alternatively ninja and clang are also supported but not described here). Depending on your linux distribution use following commands to install these tools:

Reviewer 2: provide a link to the homepages of ninja clang gcc

  ```sh
  sudo apt-get install cmake git build-essential
  ```

  Or on RPM (Red Hat Package Manager) based systems (like Fedora, openSUSE, CentOS etc.):

  ```sh
  sudo yum install -y cmake git gcc-c++
  ```

>   Or on macOS Sierra, most of the build tools can be obtained by installing Xcode (from the App Store). Other required tools may be installed using [brew ](https://brew.sh/). First install brew as described on their website. Then issue the following command to get cmake to complete the basic requirements:

Reviewer 2: link to xcode website

  ```sh
  brew install cmake git
  ```

- Installation

  If you meet all of the software requirements you can get the source code of Elektra by using this command:

  ```sh
  git clone https://github.com/ElektraInitiative/libelektra.git
  ```

>   Run the following commands to compile Elektra with non-experimental parts where your system happens to fulfil the dependences:

Reviewer 2: ```suggestion
  Run the following commands to compile Elektra with non-experimental plugins where your system happens to fulfill the dependencies:
```

  ```sh
  cd libelektra  #navigate to libelektra
  mkdir build  && cd build  #create and navigate to the build directory
  cmake ..  # watch output to see if everything needed is included

>   ccmake .. # optional: overview of the available build settings (needs cmake-curses-gui)

Reviewer 2: ```suggestion
#  optionally run "ccmake .." to get an overview of the available build settings (needs cmake-curses-gui)
```

People copy paste such stuff (even though they shouldn't). This command interrupts the flow and might confuse those who are unfarmiliar with ccmake

  make -j 5  # running makefile (compiling, linking etc.), -j specifies the number of jobs to run simultaneously

>   make run_nokdbtests  # optional: run tests

Reviewer 2: Testing is a major pain for me and my system of elektra. Especially which commands have to be run as root in order to work correctly. I reconfigured my system build process (along with I think most developers here) to avoid such issues. You could link doc\TESTING.md in a sentence stating the difficulties on testing

Reviewer: I fully agree that there needs to be a good description of what needs to happen as root. Actually it is very simple: everything that writes to spec or system namespaces needs root. E.g. `kdb mount` writes to system, and because of that it needs root.

  ```

>   With these commands you will be able to run the "Hello World!" example but usually you will need to use some of the plugins, tools and bindings of Elektra. Please take a look to the more detailed [compiling documentation](/doc/COMPILE.md). After you completed building Elektra on your own, you can execute these commands to install Elektra (please check the [installation documentation](/doc/INSTALL.md) if you are using Debian):

Reviewer 2: Is there a overview about all available plugins somewhere @markus2330 ? It would be nice to link it to the "plugins" keyword so users can have an impression of what actually exists and they might need

Reviewer 2: Maybe doc\BIGPICTURE.md?

Reviewer: The overview to plugins is in www.libelektra.org/plugins

Reviewer: (link to /src/plugins in markdown)
